src/graph/GoExecutor.cpp

```diff
@@ -530,7 +554,15 @@ std::unique_ptr<InterimResult> GoExecutor::setupInterimResult(RpcResponse &&rpcR
                     LOG(FATAL) << "Unknown VariantType: " << column.which();
             }
         }
-        rsWriter->addRow(writer);
+        std::string encode = writer.encode();
+        if (distinct_) {
+            if (uniqResult->count(encode) == 0) {
+                uniqResult->emplace(encode);
```

Comment:
You could have just called `emplace` for once, since it will tell you whether the value exists.